habitat-lab/habitat/tasks/rearrange/articulated_agent_manager.py

```diff
@@ -0,0 +1,197 @@
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Iterator, List, Optional
+
+import magnum as mn
+import numpy as np
+
+from habitat.articulated_agents.mobile_manipulator import MobileManipulator
+
+# flake8: noqa
+from habitat.articulated_agents.robots import (
+    FetchRobot,
+    FetchRobotNoWheels,
+    SpotRobot,
+    StretchRobot,
+)
+from habitat.articulated_agents.robots.fetch_suction import FetchSuctionRobot
+from habitat.tasks.rearrange.rearrange_grasp_manager import (
+    RearrangeGraspManager,
+)
+from habitat.tasks.rearrange.utils import IkHelper, is_pb_installed
+
+if TYPE_CHECKING:
+    from omegaconf import DictConfig
+
+
+@dataclass
+class ArticulatedAgentData:
+    """
+    Data needed to manage an agent instance.
+    """
+
+    articulated_agent: MobileManipulator
+    grasp_mgrs: List[RearrangeGraspManager]
```

Comment:
Where is grap_mgrs used ? It feels really close to grasp_mgr prpert. Maybe we should make it private ?